### ISSUE TITLE
Color slider for lights

### ISSUE THREAD

**Issue by FerretMonger** [AnthonMS/my-cards#20]
I would like to use the slider for controlling color values for lights.

RGB
Hue
Saturation
etc...

**Comment by AnthonMS** [comment]
Yeah, I have thought about it. But how would it work? Would you have 3 sliders, one for each color Red, Green & Blue?

**Comment by FerretMonger** [comment]
I can see multiple versions.

Either just one per "channel" (RGB) or one for Hue and one for Saturation.

It's always a bit messy to deal with colored lights. You want Brightness ofc, and you want color, but you also sometimes want to desaturate the color a bit, which means that you end up with a lot of sliders for one light or group, which in turn makes the dashboard become very bloated.

If I could choose I would want the option to choose if the slider should control brightness, hue or saturation. That way i could set up the brightness to be a large slider, and then maybe put hue and saturation as two smaller sliders underneath, for instance.

**Comment by AnthonMS** [comment]
Yeah I have had the same dilemma when thinking about setting it up for RGB. But I could set the type up to use brightness, hue and saturation instead of just brightness and warmth.

I will look into it. There are some problems with warmth slider already. Maybe I can fix these issues simultaneously. I don't know when though. There are a lot of stuff I would like to understand fully in the libraries that I use, so I can start using templating in the config.

**Comment by FerretMonger** [comment]
Sounds great! I'll take whatever I can get ❤️

**Comment by AndyVRD** [comment]
You can make a pop-up with multiple sliders. 
![8D51B6B1-8DA6-4F24-90B8-B79835EA40D8](https://user-images.githubusercontent.com/67138158/166134202-7c03287b-b5d3-451c-9b7b-6872ce7cdc6e.png)

![246E5A79-752C-4272-ACAD-ACFB87BC7742](https://user-images.githubusercontent.com/67138158/166134207-735a280d-2fd6-47ca-8cf0-fdffccb429ed.png)

![35E52D61-03FA-4C31-82B4-8A1C9C619FC9](https://user-images.githubusercontent.com/67138158/166134212-68e35f7c-2227-4a10-9a25-c3f3f063813c.png)

**Comment by AnthonMS** [comment]
Added in v2.3.0-beta.2. There's a new mode: rgb that turns a light slider into a hue picker: the track is a rainbow, and dragging the thumb sets the color.

```yaml
type: custom:my-slider-v2
entity: light.living_room
mode: rgb
```

It writes the hue at full saturation (matching the rainbow). It's the same color-track system as the temperature picker, so styles: track/progress/thumb overrides work the same way. Let me know how it looks on your setup!
